Spigot-API-Patches/0258-Implement-Keyed-on-World.patch

```diff
@@ -0,0 +1,163 @@
++    NamespacedKey getKey();
+     // Paper end
+ 
+     /**
+diff --git a/src/main/java/org/bukkit/WorldCreator.java b/src/main/java/org/bukkit/WorldCreator.java
+index 6e6945dd4c770be04ec09da3958fae751717527a..e9f603f3a6e85044191108fcab02911acd9fe64d 100644
+--- a/src/main/java/org/bukkit/WorldCreator.java
++++ b/src/main/java/org/bukkit/WorldCreator.java
+@@ -11,6 +11,7 @@ import org.jetbrains.annotations.Nullable;
+  * Represents various types of options that may be used to create a world.
+  */
+ public class WorldCreator {
++    private final NamespacedKey key; // Paper
+     private final String name;
+     private long seed;
+     private World.Environment environment = World.Environment.NORMAL;
+@@ -26,13 +27,67 @@ public class WorldCreator {
+      * @param name Name of the world that will be created
+      */
+     public WorldCreator(@NotNull String name) {
+-        if (name == null) {
+-            throw new IllegalArgumentException("World name cannot be null");
+-        }
++        // Paper start
++        this(name, NamespacedKey.minecraft(name));
++    }
+ 
+-        this.name = name;
++    /**
++     * Creates an empty WorldCreator for the given world name and key
++     *
++     * @param levelName LevelName of the world that will be created
++     * @param worldKey NamespacedKey of the world that will be created
++     */
++    public WorldCreator(@NotNull String levelName, @NotNull NamespacedKey worldKey) {
++        if (levelName == null || worldKey == null) {
++            throw new IllegalArgumentException("World name and key cannot be null");
++        }
++        this.name = levelName;
+         this.seed = (new Random()).nextLong();
++        this.key = worldKey;
++    }
++
++    /**
++     * Creates an empty WorldCreator for the given key.
++     * LevelName will be the Key part of the NamespacedKey.
++     *
++     * @param worldKey NamespacedKey of the world that will be created
++     */
++    public WorldCreator(@NotNull NamespacedKey worldKey) {
++        this(worldKey.getKey(), worldKey);
++    }
++
++    /**
++     * Gets the key for this WorldCreator
++     *
++     * @return the key
++     */
++    @NotNull
++    public NamespacedKey key() {
++        return key;
++    }
++
++    /**
++     * Creates an empty WorldCreator for the given world name and key
++     *
++     * @param levelName LevelName of the world that will be created
++     * @param worldKey NamespacedKey of the world that will be created
++     */
++    @NotNull
++    public static WorldCreator nameAndKey(@NotNull String levelName, @NotNull NamespacedKey worldKey) {
++        return new WorldCreator(levelName, worldKey);
++    }
++
++    /**
++     * Creates an empty WorldCreator for the given key.
++     * LevelName will be the Key part of the NamespacedKey.
++     *
++     * @param worldKey NamespacedKey of the world that will be created
++     */
++    @NotNull
++    public static WorldCreator key(@NotNull NamespacedKey worldKey) {
```

Comment:
maybe name those static factory methods `of` to make it obvious what they do?

Reply:
Sure